Packs/OracleCloudInfrastructure/Integrations/OracleCloudInfrastructureEventCollector/OracleCloudInfrastructureEventCollector.py

```diff
@@ -0,0 +1,396 @@
+        next_page (str | None, optional): next page query parameter for pagination. Defaults to None.
+
+    Returns:
+        requests.Response: raw response from the API.
+    """
+    params = {
+        'compartmentId': client.compartment_id,
+        'startTime': start_time,
+        'endTime': datetime.now().strftime(DATE_FORMAT)}
+    if next_page:
+        params['opc-next-page'] = next_page
+    return client._http_request(method='GET', params=params, resp_type='response')
+
+
+def add_millisecond_to_timestamp(timestamp: str) -> str:
+    """Add 1 millisecond to the given timestamp.
+
+    Args:
+        timestamp (str): Timestamp to add 1 millisecond to.
+
+    Raises:
+        DemistoException: If datetime conversion fails.
+
+    Returns:
+        str: Timestamp with 1 millisecond added.
+    """
+    try:
+        timestamp_datetime = arg_to_datetime(arg=timestamp, settings={'RETURN_AS_TIMEZONE_AWARE': False})
+        if isinstance(timestamp_datetime, datetime):
+            return (timestamp_datetime + timedelta(milliseconds=1)).strftime(DATE_FORMAT)
+        else:
+            raise DemistoException('Datetime conversion failed.')
+    except Exception as e:
+        raise DemistoException(message=e) from e
+
+
+def get_events(
+        client: Client, first_fetch_time: datetime, max_fetch: int) -> tuple[
+        list[dict[str, Any]],
+        str]:
+    """Get events from an oracle cloud infrastructure tenant.
+    - The request returns a maximum of 100 events per call by default.
+    - This function uses pagination, meaning it will make multiple request as needed to reach the desired amount of events.
+
+    Args:
+        client (Client): Client object for requests.
+        first_fetch_time (datetime): The start time to fetch events from.
+        max_fetch (int): The limit of events to fetch.
+
+    Raises:
+        DemistoException: If an error occurred while fetching events.
+
+    Returns:
+        tuple[list[dict[str, Any]], str]: A tuple of the events list and the last event time for next fetch cycle.
+    """
+    try:
+        response = audit_log_api_request(client=client, start_time=first_fetch_time.strftime(DATE_FORMAT))
+        events = json.loads(response.content)
+
+        if not events:
+            return [], first_fetch_time.strftime(DATE_FORMAT)
+
+        if isinstance(events, dict):
+            events = [events]
+
+        # pagination handling
+        while len(events) < max_fetch and (next_page := response.headers._store.get('opc-next-page')):  # type: ignore
+            current_start_time = add_millisecond_to_timestamp(events[-1].get('eventTime'))
+            response = audit_log_api_request(client=client, start_time=current_start_time, next_page=next_page[1])
+            events.extend(json.loads(response.content))
+
+        last_event_time = get_last_event_time(events, first_fetch_time)
+        events = add_time_key_to_events(events)
+
+    except Exception as e:
```

Comment:
Let's add a comment explaining what we are doing in this block.

Reply:
Done.